gfx_scene.h

```diff
@@ -1744,35 +1744,47 @@ class GfxSceneInternal
                     TransformGltfLight(*light, transform);
                 }
             }
-            bool is_any_children_animated = false;
-            std::map<int32_t, uint64_t>::const_iterator const it = animated_nodes.find(node_index);
+            std::map<int32_t, std::pair<size_t, uint64_t>>::const_iterator const it = animated_nodes.find(node_index);
             bool is_node_animated = (is_parent_animated || it != animated_nodes.end());
+            size_t node_animation = (is_parent_animated ? parent_animation : (it != animated_nodes.end() ? (*it).second.first : std::numeric_limits<size_t>::max()));
+            bool is_any_children_animated = false;
+            size_t children_animation = std::numeric_limits<size_t>::max();
             for(size_t i = 0; i < gltf_node.children.size(); ++i)
-                is_any_children_animated = (VisitNode(gltf_node.children[i], transform, is_node_animated) || is_any_children_animated);
-            is_node_animated = (is_node_animated || is_any_children_animated);
+            {
+                auto const [is_child_animated, child_animation] = VisitNode(gltf_node.children[i], transform, is_node_animated, node_animation);
+                GFX_ASSERT(!is_any_children_animated || !is_child_animated || children_animation == child_animation); // we assume non overlapping animations
+                is_any_children_animated = is_any_children_animated || is_child_animated;
+                children_animation = is_child_animated ? child_animation : children_animation;
+            }
+            if (!is_node_animated && is_any_children_animated)
+            {
+                is_node_animated = true;
+                node_animation = children_animation;
+            }
```

Comment:
We propagate animations through the entire node hierarchy (from roots to leaves). We ensure each animated node (including parent nodes and children) is affected by one animation (no overlap between animation).